app/src/providerrails/java/com/layer/messenger/flavor/util/CustomEndpoint.java

```
@@ -112,43 +112,48 @@ private static Endpoint getEndpoint() {
+                    writer.write(buffer, 0, n);
+                }
+            } catch (Exception e) {
+                if (Log.isLoggable(Log.ERROR)) Log.e(e.getMessage(), e);
+            } finally {
+                if (is != null) {
+                    try {
+                        is.close();
+                    } catch (IOException e) {
+                        if (Log.isLoggable(Log.ERROR)) Log.e(e.getMessage(), e);
+                    }
+                }
             }
-        } catch (Exception e) {
-            if (Log.isLoggable(Log.ERROR)) Log.e(e.getMessage(), e);
-        } finally {
-            if (is != null) {
-                try {
-                    is.close();
-                } catch (IOException e) {
-                    if (Log.isLoggable(Log.ERROR)) Log.e(e.getMessage(), e);
+            String content = writer.toString().trim();
+            if (content.isEmpty()) return null;
+
+            // Parse endpoints from JSON
+            try {
+                JSONArray array = new JSONArray(content);
+                for (int i = 0; i < array.length(); i++) {
+                    Endpoint endpoint = new Endpoint(array.getJSONObject(i));
+                    sEndpoints.put(endpoint.getName(), endpoint);
                 }
+                return sEndpoints;
+            } catch (JSONException e) {
+                if (Log.isLoggable(Log.ERROR)) Log.e(e.getMessage(), e);
             }
-        }
-        String content = writer.toString().trim();
-        if (content.isEmpty()) return null;
 
-        // Parse endpoints from JSON
-        try {
-            JSONArray array = new JSONArray(content);
-            for (int i = 0; i < array.length(); i++) {
-                Endpoint endpoint = new Endpoint(array.getJSONObject(i));
-                sEndpoints.put(endpoint.getName(), endpoint);
-            }
-            return sEndpoints;
-        } catch (JSONException e) {
+        } catch (IOException e) {
```

Crash or handle! Don't ever swallow exceptions :)